pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt

```diff
@@ -2,159 +2,197 @@ package gov.cdc.ocio.processingstatusapi.plugins
+                            val deleteMessageRequest = DeleteMessageRequest {
+                                this.queueUrl = queueUrl
+                                this.receiptHandle = message.receiptHandle
+                            }
+                            sqsClient.deleteMessage(deleteMessageRequest)
+                        }
+                        SchemaValidation.logger.info(
+                                "Successfully deleted processed report from AWS SQS"
+                        )
+                    } catch (e: Exception) {
+                        SchemaValidation.logger.error(
+                                "Something went wrong while deleting the report from the queue ${e.message}"
+                        )
+                    }
                 }
             }
-            deleteMessage(receivedMessages)
-        }catch (e: Exception) {
-            SchemaValidation.logger.error("An Exception occurred during validation ${e.message}")
-            deleteMessage(receivedMessages)
-        }
-    }
-    /**
-     * The `consumeMessages` function continuously listens for and processes messages from an AWS SQS queue.
-     * This function runs in a non-blocking coroutine, retrieving messages from the queue, validating them using
-     * `AWSSQSProcessor`, and then deleting the processed messages from the queue.
-     *
-     * @throws Exception
-     * @throws AwsServiceException
-     */
-    fun consumeMessages() {
-        SchemaValidation.logger.info("Consuming messages from AWS SQS")
-        CoroutineScope(Dispatchers.IO).launch {
-            while (true) {
-                var receivedMessages: ReceiveMessageResponse?
+            /**
+             * Validates messages from the AWS SQS Service
+             * @param receivedMessages the list of message(s) received from the queue to be
+             * validated
+             * @throws Exception thrown during validation `and` it's important to delete the message
+             * as it will be persisted to dead-letter container
+             * ```
+             *                    in configured database
+             * ```
+             */
+            suspend fun validate(receivedMessages: ReceiveMessageResponse) {
                 try {
-                    receivedMessages = retryWithBackoff(numOfRetries = 5){
-                        val receiveMessageRequest = ReceiveMessageRequest {
-                            this.queueUrl = queueUrl
-                            maxNumberOfMessages = 5
+                    receivedMessages.messages?.forEach { message ->
+                        SchemaValidation.logger.info(
+                                "Received message from AWS SQS: ${message.body}"
+                        )
+                        val awsSQSProcessor = AWSSQSProcessor()
+                        message.body?.let {
+                            val rawMessage = components.gson.fromJson(it, SQSMessage::class.java)
```

Comment:
Is there anyway from the message headers or metadata to determine if the message is raw or base64 encoded?  If so, would love to support either.  If everything needs to be base64 encoded then we need to update our load testing script to encode all the reports it sends.  Not a big deal, but would love to support either and figure out which one it is programmatically at runtime.

Reply:
good catch, i'm going to make this unnecessary, but i do think there's a content type attribute for messages